### PR TITLE
conditional skip Doctests

### LINKED ISSUE #733: Improve doctests - conditional dependency skip
## 🚀 Feature

use conditional skip for doctest which are showing metrics with additional dependency
Thx @ethanwharris for finding this nice pytest extension - `pytest-doctestplus`
see sample usage: https://github.com/astropy/pytest-doctestplus#doctest-dependencies

### Motivation

have all doctests/examples tested again

### Pitch

At this moment we are skipping all which are not pure PyTorch so the outputs could be wrong after several changes in codebase

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes #733

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@stancld seems the docs example does not work as BERT requires models name... ^^

File: torchmetrics/functional/text/bert.py [hunk 1]
```diff
@@ -566,7 +568,7 @@ def bert_score(
         >>> from torchmetrics.functional.text.bert import bert_score
         >>> preds = ["hello there", "general kenobi"]
         >>> target = ["hello there", "master kenobi"]
-        >>> bert_score(preds, target, lang="en")  # doctest: +SKIP
+        >>> bert_score(preds, target, lang="en")
```
Comment (stancld) [inline torchmetrics/functional/text/bert.py:571]:
```suggestion
        >>> bert_score(preds, target, model_name_or_path="roberta-base")
```

File: torchmetrics/text/bert.py [hunk 2]
```diff
@@ -115,8 +117,8 @@ class BERTScore(Metric):
         >>> from torchmetrics.text.bert import BERTScore
         >>> preds = ["hello there", "general kenobi"]
         >>> target = ["hello there", "master kenobi"]
-        >>> bertscore = BERTScore()
-        >>> bertscore(preds, target)  # doctest: +SKIP
+        >>> bert = BERTScore()
```
Comment (stancld) [inline torchmetrics/text/bert.py:120]:
```suggestion
        >>> bert = BERTScore(model_name_or_path="roberta-base")
```

**Review by stancld — COMMENTED** [review]
Let's try to add `model_name_or_path="roberta-base"` manually if it works :]

**Comment by Borda** [comment]
> Let's try to add `model_name_or_path="roberta-base"` manually if it works :]

so it means that default none is not valid input...

**Comment by stancld** [comment]
> > Let's try to add `model_name_or_path="roberta-base"` manually if it works :]
> 
> so it means that default none is not valid input...

Unfortunately, it looks like so.. I'll send a quick fix tomorrow morning. It should be only about some input validation.

File: torchmetrics/text/bert.py [hunk 3]
```diff
@@ -115,18 +117,18 @@ class BERTScore(Metric):
...
-        >>> bertscore = BERTScore()
-        >>> bertscore(preds, target)  # doctest: +SKIP
-        {'precision': [0.99..., 0.99...],
-         'recall': [0.99..., 0.99...],
-         'f1': [0.99..., 0.99...]}
+        >>> bert = BERTScore(model_name_or_path="roberta-base")
+        >>> bert(preds, target)  # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+        {'precision': [0.999..., 0.999., 0.987..., 0.987...],
+         'recall': [0.999..., 0.999..., 0.987..., 0.987...],
+         'f1': [0.999..., 0.999..., 0.987..., 0.987...]}
```
Comment (stancld) [inline torchmetrics/text/bert.py:124]: This looks quite odd to me as the length of output should be 2 instead of 4.. gonna have a look

Comment (Borda) [reply torchmetrics/text/bert.py:124]: so it is good that we have these examples tested again... :rabbit:

File: torchmetrics/text/bert.py [hunk 4]
```diff
@@ -130,7 +132,7 @@ class BERTScore(Metric):
 
     def __init__(
         self,
-        model_name_or_path: Optional[str] = None,
+        model_name_or_path: str,
```
Comment (stancld) [inline torchmetrics/text/bert.py:135]:
```suggestion
        model_name_or_path: Optional[str] = None,
```

After the fix at #760, `model_name_or_path` argument should remain optional

File: torchmetrics/functional/text/bert.py [hunk 5]
```diff
@@ -465,7 +467,7 @@ def _rescale_metrics_with_baseline(
 def bert_score(
     preds: Union[List[str], Dict[str, Tensor]],
     target: Union[List[str], Dict[str, Tensor]],
-    model_name_or_path: Optional[str] = None,
+    model_name_or_path: str,
```
Comment (stancld) [inline torchmetrics/functional/text/bert.py:470]: ^^ ops, same here :]
```suggestion
    model_name_or_path: Optional[str] = None,
```

File: torchmetrics/detection/map.py [hunk 6]
```diff
@@ -26,10 +26,20 @@
...
 
+__doctest_requires__ = {
+    ("ROUGEScore", "rouge_score", "_rouge_score_update"): ["nltk"],
+    ("BERTScore", "bert_score"): ["transformers"],
+    ("FrechetInceptionDistance", "FID"): ["torch_fidelity"],
+    ("KernelInceptionDistance", "KID"): ["torch_fidelity"],
+    ("InceptionScore", "IS"): ["torch_fidelity"],
+}
+
+
```
Comment (stancld) [inline torchmetrics/detection/map.py:42]: Should this file `torchmetriscs/detection/map.py` contain these `doctest` instructions?

File: .github/workflows/ci_test-base.yml [hunk 7]
```diff
@@ -52,27 +51,20 @@ jobs:
...
         pip uninstall -y torchmetrics
         pip list
       shell: bash
 
-    - name: HF cache
-      uses: actions/cache@v2
-      with:
-        path: $TRANSFORMERS_CACHE
-        key: cache-transformers
-
```
[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/ci_test-base.yml:75]: it is just in this workflow where we install only the required dependencies not all domains

File: torchmetrics/detection/map.py (hunk 6 above)
Comment (Borda) [inline torchmetrics/detection/map.py:42]:
```suggestion
```

File: torchmetrics/detection/map.py [hunk 8]
```diff
@@ -26,10 +26,20 @@
     from torchvision.ops import box_area, box_convert, box_iou
 else:
     box_convert = box_iou = box_area = None
+    __doctest_skip__ = ["MeanAveragePrecision", "MAP"]
```
Comment (Borda) [inline torchmetrics/detection/map.py:29]:
```suggestion
```

File: torchmetrics/detection/map.py (hunk 6 above)
Comment (Borda) [reply torchmetrics/detection/map.py:42]: not necessary as the minimal package mostly uses latest versions

**Review by justusschock — APPROVED** [review]

**Review by stancld — APPROVED** [review]
⭐ LGTM!

**Comment by Borda** [comment]
the HF cache seems strange as I still see time to time following download error:
```
>           raise HTTPError(http_error_msg, response=self)
E           requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://huggingface.co/albert-base-v2/resolve/main/spiece.model
```
(mostly on MacOS)

**Pull request merged**